source/lambda/use-case-management/model/use-case.ts

```diff
@@ -296,25 +297,47 @@ export class ChatUseCaseDeploymentAdapter extends UseCase {
         );
         cfnParameters.set(CfnParameterKeys.UseCaseConfigTableName, process.env[USE_CASE_CONFIG_TABLE_NAME_ENV_VAR]!);
 
-        cfnParameters.set(CfnParameterKeys.ExistingCognitoUserPoolId, process.env[USER_POOL_ID_ENV_VAR]!);
-        if (
-            process.env[USE_EXISTING_USER_POOL_CLIENT_ENV_VAR] &&
-            process.env[USE_EXISTING_USER_POOL_CLIENT_ENV_VAR].toLowerCase() === 'true' &&
-            process.env[CLIENT_ID_ENV_VAR]
-        ) {
-            cfnParameters.set(CfnParameterKeys.ExistingCognitoUserPoolClient, process.env[CLIENT_ID_ENV_VAR]);
+        if (eventBody.AuthenticationParams) {
+            switch (eventBody.AuthenticationParams.AuthenticationProvider) {
+                case AUTHENTICATION_PROVIDERS.COGNITO:
+                    const existingUserPoolId = eventBody.AuthenticationParams.CognitoParams.ExistingUserPoolId;
+                    const existingUserPoolClientId = eventBody.AuthenticationParams.CognitoParams.ExistingUserPoolClientId;
+
+                    if (!existingUserPoolId) {
+                        throw new Error(`Required field existingUserPoolId not provided for the "Cognito" AuthenticationProvider.`);
+                    }
+
+                    cfnParameters.set(CfnParameterKeys.ExistingCognitoUserPoolId, existingUserPoolId);
+
+                    if (existingUserPoolClientId) {
+                        cfnParameters.set(CfnParameterKeys.ExistingCognitoUserPoolClient, existingUserPoolClientId);
+                    }
+
+                    break;
+            }
         }
+        else {
+            cfnParameters.set(CfnParameterKeys.ExistingCognitoUserPoolId, process.env[USER_POOL_ID_ENV_VAR]!);
+
+            if (
+                process.env[USE_EXISTING_USER_POOL_CLIENT_ENV_VAR] &&
+                process.env[USE_EXISTING_USER_POOL_CLIENT_ENV_VAR].toLowerCase() === 'true' &&
+                process.env[CLIENT_ID_ENV_VAR]
+            ) {
+                cfnParameters.set(CfnParameterKeys.ExistingCognitoUserPoolClient, process.env[CLIENT_ID_ENV_VAR]);
+            }
```

Comment:
we discussed this functionality on our end when working on support for custom pools in the deployment dashboard, and elected to remove the `USE_EXISTING_USER_POOL_CLIENT_ENV_VAR` variable. It was not being set anywhere or used in our code.

If you wanted to use an existing client, you could now with this new functionality pass the client in the API call under ExistingCognitoUserPoolClient. It is also a weird edge case to want to use the same client as the deployment dashboard for your use cases.

As such, this highlighted if block can be removed.